File: algorithm/python/BOJ10818.py
# problem : https://www.acmicpc.net/problem/10818
import sys

N = int(sys.stdin.readline().rstrip())
arr = list(map(int, sys.stdin.readline().rstrip().split()))
min_value, max_value = sys.maxsize, -sys.maxsize
for num in arr:
    if num < min_value:
        min_value = num
    if num > max_value:
        max_value = num

print(min_value, max_value)

File: algorithm/java/src/BOJ10818.java
// problem : https://www.acmicpc.net/problem/10818

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class BOJ10818 {
    static int N, min = Integer.MAX_VALUE, max = Integer.MIN_VALUE;
    static int[] arr;
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        arr = new int[N];

        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
            if (arr[i] < min) {
                min = arr[i];
            }

            if (arr[i] > max) {
                max = arr[i];
            }
        }

        // Arrays.sort(arr);
        // System.out.println(arr[0] + " " + arr[N - 1]);
        System.out.println(min + " " + max);
    }
}
